test/integration/targets/meta_tasks/runme.sh

```diff
@@ -14,10 +14,17 @@ done
 
 # test end_host meta task, on all hosts
 for test_strategy in linear free; do
-  out="$(ansible-playbook test_end_host_all.yml -i inventory -e test_strategy=$test_strategy -vv "$@")"
+  out="$(ansible-playbook test_end_host_all.yml -i inventory.yml -e test_strategy=$test_strategy -vv "$@")"
 
   grep -q "META: ending play for testhost" <<< "$out"
   grep -q "META: ending play for testhost2" <<< "$out"
   grep -qv "play not ended for testhost" <<< "$out"
   grep -qv "play not ended for testhost2" <<< "$out"
 done
+
+# test end_play meta task
+for test_strategy in linear free; do
+  out="$(ansible-playbook test_end_play.yml -i inventory.yml -e test_strategy=$test_strategy -vv "$@")"
+
+  grep -q "META: ending play" <<< "$out"
```

Comment:
Add a check to make sure the `Failed to end` message isn't present.

Reply:
I would expect the non-zero exit code from fail to detect this, but it's probably worth being explicit about the test.

Reply:
I did check and the test will report a failure if the `fail` task is triggered. I'll add this as another check, though. More checks are always better. 😄